src/db_connection.ts

```diff
@@ -2,9 +2,10 @@ import { Connection, createConnection } from 'typeorm';
 
 import { config } from './ormconfig';
 
+const connection = createConnection(config);
 /**
  * Creates the DB connnection to use in an app
  */
 export async function getDBConnectionAsync(): Promise<Connection> {
-    return createConnection(config);
+    return connection;
```

Comment:
Why this change?

Reply:
`createConnection` actually creates a connection pool. It's intended to be called once per app and continue running in the background. Previous code was creating connection pools every time this method is called.

https://github.com/typeorm/typeorm/blob/master/docs/connection.md

Reply:
Yea but the method was only being called once.  I just don't want to encourage importing this and calling it directly, but rather calling it once and dependency injecting.

Reply:
It's kind of confusing to export a function that calls another function that's intended to be only called once. Exporting it encourages it to be called anywhere else.

This way `createConnection` **can** only be called once, and exporting the function as a getter encourages it to be used anywhere else in the app that a connection is needed.